pkg/agent/datamodel/sig_config.go

```diff
@@ -58,106 +58,36 @@ func GetCloudTargetEnv(location string) string {
 	}
 }
 
-/*
-AvailableUbuntu1804Distros : TODO(amaheshwari): these vars are not consumed by Agentbaker but by RP. do a
-cleanup to remove these after 20.04 work.
-*/
-var AvailableUbuntu1804Distros []Distro = []Distro{
-	AKSUbuntu1804,
-	AKSUbuntu1804Gen2,
-	AKSUbuntuGPU1804,
-	AKSUbuntuGPU1804Gen2,
-	AKSUbuntuContainerd1804,
-	AKSUbuntuContainerd1804Gen2,
-	AKSUbuntuGPUContainerd1804,
-	AKSUbuntuGPUContainerd1804Gen2,
-	AKSUbuntuFipsContainerd1804,
-	AKSUbuntuFipsContainerd1804Gen2,
-	AKSUbuntuFipsGPUContainerd1804,
-	AKSUbuntuFipsGPUContainerd1804Gen2,
-	AKSUbuntuArm64Containerd1804Gen2,
-	AKSUbuntuEdgeZoneContainerd1804,
-	AKSUbuntuEdgeZoneContainerd1804Gen2,
-}
-
-var AvailableUbuntu2004Distros []Distro = []Distro{
-	AKSUbuntuContainerd2004CVMGen2,
-	AKSUbuntuFipsContainerd2004,
-	AKSUbuntuFipsContainerd2004Gen2,
-}
-
-var AvailableUbuntu2204Distros []Distro = []Distro{
-	AKSUbuntuContainerd2204,
-	AKSUbuntuContainerd2204Gen2,
-	AKSUbuntuArm64Containerd2204Gen2,
-	AKSUbuntuContainerd2204TLGen2,
-}
-
-var AvailableContainerdDistros []Distro = []Distro{
-	AKSUbuntuContainerd1804,
-	AKSUbuntuContainerd1804Gen2,
-	AKSUbuntuGPUContainerd1804,
-	AKSUbuntuGPUContainerd1804Gen2,
-	AKSUbuntuFipsContainerd1804,
-	AKSUbuntuFipsContainerd1804Gen2,
-	AKSUbuntuFipsContainerd2004,
-	AKSUbuntuFipsContainerd2004Gen2,
-	AKSUbuntuFipsGPUContainerd1804,
-	AKSUbuntuFipsGPUContainerd1804Gen2,
-	AKSUbuntuEdgeZoneContainerd1804,
-	AKSUbuntuEdgeZoneContainerd1804Gen2,
-	AKSCBLMarinerV1,
-	AKSCBLMarinerV2,
-	AKSCBLMarinerV2Gen2,
-	AKSCBLMarinerV2Gen2Kata,
-	AKSCBLMarinerV2Gen2TL,
-	AKSCBLMarinerV2KataGen2TL,
-	AKSUbuntuArm64Containerd1804Gen2,
-	AKSUbuntuArm64Containerd2204Gen2,
-	AKSUbuntuContainerd2204,
-	AKSUbuntuContainerd2204Gen2,
-	AKSUbuntuContainerd2004CVMGen2,
-	AKSUbuntuContainerd2204TLGen2,
-}
-
-var AvailableGPUDistros []Distro = []Distro{
-	AKSUbuntuGPU1804,
-	AKSUbuntuGPU1804Gen2,
-	AKSUbuntuGPUContainerd1804,
-	AKSUbuntuGPUContainerd1804Gen2,
-	AKSUbuntuFipsGPUContainerd1804,
-	AKSUbuntuFipsGPUContainerd1804Gen2,
-}
-
-var AvailableGen2Distros []Distro = []Distro{
-	AKSUbuntu1804Gen2,
-	AKSUbuntuGPU1804Gen2,
-	AKSUbuntuContainerd1804Gen2,
-	AKSUbuntuGPUContainerd1804Gen2,
-	AKSUbuntuFipsContainerd1804Gen2,
-	AKSUbuntuFipsContainerd2004Gen2,
-	AKSUbuntuFipsGPUContainerd1804Gen2,
-	AKSUbuntuEdgeZoneContainerd1804Gen2,
-	AKSUbuntuArm64Containerd1804Gen2,
-	AKSUbuntuArm64Containerd2204Gen2,
-	AKSUbuntuContainerd2204Gen2,
-	AKSUbuntuContainerd2004CVMGen2,
-	AKSUbuntuContainerd2204TLGen2,
-}
-
-var AvailableCBLMarinerDistros []Distro = []Distro{
-	AKSCBLMarinerV1,
-	AKSCBLMarinerV2,
-	AKSCBLMarinerV2Gen2,
-	AKSCBLMarinerV2Gen2Kata,
-	AKSCBLMarinerV2Arm64Gen2,
-	AKSCBLMarinerV2Gen2TL,
-	AKSCBLMarinerV2KataGen2TL,
-}
-
 // IsContainerdSKU returns true if distro type is containerd-enabled.
 func (d Distro) IsContainerdDistro() bool {
-	for _, distro := range AvailableContainerdDistros {
+	availableContainerdDistros := []Distro{
```

Comment:
can we keep these as variables but make them unexported instead of redefining per func?

if not this is okay

Reply:
Yeah making it unexported still violates the no global variables rule